### ISSUE TITLE
ffmpeg recipe possibly broken

### ISSUE THREAD

**Issue by germn** [kivy/python-for-android#920]
I'm trying to build simple possible app using video widget with buildozer.

`requirements = hostpython2, kivy==1.9.1, ffmpeg`

`buildozer android_new debug`

Error I'm getting:

> ./libpng_build.sh: line 17: ./configure: No such file or directory

Ubuntu 16.04 in VirtualBox, everything installed by instructions on site.

**Comment by inclement** [comment]
Just to check, this does not happen if you remove ffmpeg from the requirements?

**Comment by germn** [comment]
On removing ffmpeg from the requirements everything builds fine (except video doesn't work on Android).

Btw, is there any alternatives to ffmpeg to use video widget on Android? I didn't find it in docs, but looks like ffmpeg is the only way now, am I right?

**Comment by inclement** [comment]
I'm not sure about this, I haven't tried video. I'll try to look into
building the recipe when I have time though, feel free to open a
python-for-android github issue about it if there isn't one already.

**Comment by germn** [comment]
Looks like problem comes from [WritingMinds/ffmpeg-android](https://github.com/WritingMinds/ffmpeg-android) used by this [recipe](https://github.com/kivy/python-for-android/blob/master/pythonforandroid/recipes/ffmpeg/__init__.py), see this [issue](https://github.com/WritingMinds/ffmpeg-android/issues/45). There's also [PR to fix it](https://github.com/WritingMinds/ffmpeg-android/pull/35) from may, but it didn't accepted yet.

Looking at project's [state](https://github.com/WritingMinds/ffmpeg-android/commits/master) it seems to be dead or poorly maintained :(

What can be done in this case?

**Comment by KeyWeeUsr** [comment]
Fork under Kivy and fix it? If there's that PR and it works, it shouldn't be hard. If the project is going to die (or is dead anyway), this could be a solution. Pinging the repo author first might be a good idea.

**Comment by germn** [comment]
I tried to build project with updated link, but got new error later:

```
cd builds/unix; \
        ./configure  '--with-pic' '--with-sysroot=/home/kivy/test/.buildozer/android/platform/build/build/other_builds/ffmpeg/armeabi-v7a/ffmpeg/toolchain-android/sysroot' '--host=arm-linux-androideabi' '--enable-static' '--disable-shared' '--with-png=yes' '--with-zlib=yes' '--prefix=/home/kivy/test/.buildozer/android/platform/build/build/other_builds/ffmpeg/armeabi-v7a/ffmpeg/toolchain-android'
/bin/sh: 2: ./configure: not found
builds/unix/detect.mk:85: recipe for target 'setup' failed
make: *** [setup] Error 127
```

I have no idea how to fix it.

**Comment by germn** [comment]
Finally, I found [this thread](https://groups.google.com/forum/#!topic/kivy-users/ypJOZXlUEqM):


> TLDR: replace ffmpeg with ffpyplayer as a requirements for video support on Android and iOS.
> 
> Forget about your previous experience for Video on Android / iOS. The previous provider was made by myself during 2012, based on ffmpeg 1.2. It was largely buggy for some cases, and performance was not great.

Problem with video on Android still exists since `ffpyplayer` have no recipe for new toolchain, see [464](https://github.com/kivy/python-for-android/issues/464).

Anyway as I understand, `ffmpeg` can be completely removed from p4a since 1) it's broken 2) it's replaced with `ffpyplayer`.

**Comment by germn** [comment]
Problem above was because ffmpeg-android need some dependences not mentioned anywhere in p4a:

> sudo apt-get --quiet --yes install build-essential git autoconf libtool pkg-config gperf gettext yasm python-lxml 

----
I'm trying to fix recipe for ffmpeg and add recipe for ffpyplayer:
https://github.com/germn/ffmpeg-android
https://github.com/germn/python-for-android

Looks like ffmpeg recipe is ok now: it compiles shared libraries without errors.

While I still can't create correct [ffpyplayer recipe](https://github.com/germn/python-for-android/blob/master/pythonforandroid/recipes/ffpyplayer/__init__.py), it fails with errors:
http://pastebin.com/dvW3anP9

I don't understand why it happens since all libraries that can not be found exists in specified FFMPEG_LIB_DIR folder. Also manual installing of ffpyplayer (via setup.py) seems to be ok.

If you have any ideas how ffpyplayer recipe should be changed, feel free to help me with this.

**Comment by inclement** [comment]
Thanks for the detailed examination. Would you be interested in doing a PR to add these requirements to the doc somewhere?

**Comment by germn** [comment]
1)
Should this requirements be added to p4a [installing dependencies](http://python-for-android.readthedocs.io/en/latest/quickstart/#installing-dependencies)? I mean since ffmpeg is only lib that need it, may be there's some way to add dependencies for concrete recipe?

2)
I of course will make PR for all this stuff when (and if) I'll be able to make it work. Right now current ffmpeg is broken anyway so just adding it's dependencies to docs wouldn't help.

I also not sure if fixed ffmpeg version is "correct": old ffmpeg recipe (when it was working) was building static libraries while fixed one builds shared (since ffpyplayer needs shared libraries). I don't understand reason of that yet and if static libs still actual.

**Comment by germn** [comment]
Current `ffmpeg` recipe can't be used to play Kivy video standard way (it's not only broken, but also compiles static ffmpeg while ffpyplayer video provider needs shared one).

I still don't understand why this recipe was added, but I think it can be removed since it's broken and confusing (ffpyplayer itself might need this recipe name for it's ffmpeg dependcy).

**Comment by SagitSri** [comment]
Then how they add ffmpeg in kivy launcher?

**Comment by germn** [comment]
@SagitSri this issue is closed because it is outdated. AFAIK ffmpeg recipe has been rewritten and is working now.

**Comment by SagitSri** [comment]
Thank you sir!
requirements = AFAIK ffmpeg
Am I right sir?

**Comment by SagitSri** [comment]
As per your guidelines i did the followings and got these errors sir! I've
attached buildozer error file and buildozer.spec files! Kindly help me to
get out from this problem sir!

**Comment by germn** [comment]
@SagitSri please consider creating [new issue](https://github.com/kivy/python-for-android/issues/new) about problem you have. It'll help other people faced with same problem and allow to find answer faster.

**Comment by SagitSri** [comment]
Thank you sir!

**Comment by hakanb78** [comment]
Hi, having this same issue with kivy 1.11.1  and python 3.7 running on ubuntu VM. Kivy app works on linux but when i add the ffpyplayer to requirements, APK build fails. Anyone can help?
